packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js

```diff
@@ -115,19 +122,25 @@ angular.module('App').controller(
       this.loading.durations = true;
 
       this.privateDatabaseService
-        .getRamPrices(this.productId, {
-          ram: this.model.capacity,
-        })
-        .then(
-          (durations) => {
-            this.loading.durations = false;
-            this.data.durations.available = durations;
-          },
-          angular.noop,
-          (duration) => {
-            this.data.durations.available = duration;
-          },
-        );
+        .getRamPlan(this.productId, this.model.ram.planCode)
+        .then((order) => {
+          this.loading.durations = false;
+
+          const details = order.details.filter((detail) => {
+            return detail.detailType === 'DURATION';
+          });
```

Comment:
```suggestion
          const details = order.details.filter((detail) => {
            return detail.detailType === 'DURATION';
          });
```
const details = order.details.filter(({detailType}) =>
detailType === 'DURATION';
);